File: 01-day/singleNumber.py
def singleNumber(nums: list[int]) -> int:
    """
    :type nums: list[int]
    :rtype: int
    """
    n = nums[0]
    for i in range(1, len(nums)):
        n = n ^ nums[i]
    return n


if __name__ == "__main__":
    nums = [2, 2, 1]
    print(f"{singleNumber(nums)}")
    nums = [4, 1, 2, 1, 2]
    print(f"{singleNumber(nums)}")

File: 01-day/singleNumber.cpp
#include <vector>
#include <iostream>

int singleNumber(std::vector<int> &nums)
{
    int n = nums[0];
    for (size_t i = 1; i < nums.size(); i++)
        n = n ^ nums[i];
    return n;
}

int main(int argc, char const *argv[])
{
    /* code */
    std::vector<int> num = {4, 1, 2, 1, 2};
    int res = singleNumber(num);
    std::cout << "singleNumber: " << res << std::endl;
    return 0;
}
